### ISSUE TITLE
Admonition plugin and also custom CSS selected in Obsidian appearence do not work in your v.1.7.3

### ISSUE THREAD

**Issue by vgyenge6** [KosmosisDire/obsidian-webpage-export#284]
Hi,

First of all, it is a great plugin. Thanks for it.

Since your plugin [v.1.6.0](https://github.com/KosmosisDire/obsidian-webpage-export/releases/tag/1.6.0) says the Admonition plugin is supported.

However, this version does not list the Admonition plugin in the Include Plugin CSS anymore. (I tried the previously mentioned version and it did it well.)

I tried to export the Admonition plugin's custom callout styles added it to CSS snippets on the global Appearance setting and enabled it. However, I noticed neither this custom callout style nor my those set in the snippets CSS globally allowed are embedded when I export a single .md file. (I also tried to copy the custom callout CSS into the snippets CSS and allowed only the snippet CSS supposing this should have to be embedded at least in the export when allowed.)

Please look after this, because at least the globally allowed CSS custom settings should have to be embedded.
Thank you.

![Settings_Special_features_Admonition_custom-styles_do_not work_2023-11-21](https://github.com/KosmosisDire/obsidian-webpage-export/assets/100508773/5196325a-97ae-4b4f-bff4-eed5562f87fd)

**Operating System**
WIN 10 Pro
Obsidian  v.1.4.16
your plugin v. 1.7.3
Admonition plugin v.10.1.1

**Comment by KosmosisDire** [comment]
Hi thanks, I beleive you may actually be experiencing issue #229. The Admonition plugin is not shown in the custom css section because it's styles are automatically included.

**Comment by vgyenge6** [comment]
Thanks for your reply. I would like to clarify my issue about the custom CSS embedding.

I checked my custom CSS properties involved in the snippets.CSS allowed in the general appearance setting is well embedded.
e.g.
img[alt*="left"] {
  float: left;
  padding-right:10px;
}
summary {
	font-size: 12px;
	color: #16A29F;
}

Am I understanding well that, however?
1. If the Admonition plugin is installed it will not appear in your plugin Include Plugin CSS settings, because adding the Admonition plugin's CSS will embed automatically.
2. Will it contain the custom styles set in the Admonition plugin 
3. The above-mentioned issue https://github.com/KosmosisDire/obsidian-webpage-export/issues/229. is it because there is a problem to output the icons only?
4. Will in this case the Admonition plugins' collapsible and copy-to-clipboard feature also work? (It doesn't work now in the HTML export)

See Admonition plugin settings and appearance in the reading view.
![Admonition_settings_2023-11-22](https://github.com/KosmosisDire/obsidian-webpage-export/assets/100508773/1cdb0e88-594e-4eca-af8b-df7ddbe56455)

**Comment by KosmosisDire** [comment]
> Will in this case the Admonition plugins' collapsible and copy-to-clipboard feature also work? (It doesn't work now in the HTML export)

No these features will not work right now. I have only added support for the styles because the admonitions plugin adds styles procedurally they could not be added from the css menu before. Which is why I had to add specific support for it. However I did not do anything to support the dynamic features, only the styles. I will think about adding this in the future

**Comment by vgyenge6** [comment]
Thank you for your answer.
Btw: Admonition Plugin now supports exporting the custom styles to a CSS file, which may be allowed for use as set in the obsidian global settings Appereance. (see the Admonition plugin's setting picture above Export Custom Types as CSS).
Also obsidian inbuilt callout supports 'collapsible' feature (without using Admonition plugin).
I guess collapsing the admonition/callout block would work by adding a simple JS or CSS resolution .
The Admonition plugin copy to clipboard feature also would be nice to have, but really not so important. :)

**Comment by KosmosisDire** [comment]
This feature has been added in 1.8.0! :)
If you find any issues please create a new issue
Thanks!

**Comment by vgyenge6** [comment]
Hi,

Thank you, it works now well. 👍

**Comment by Chupakabra0** [comment]
Seems to me still issue when I use old admonition syntax in v.1.8.0.
---
Obsidian downgraded to 1.5.3 after 1.5.8 "had broken everything"
OS: Win10 Pro
Admonition plugin 10.2.0.
My custom .css and theme didn't affect anything.

## Example:
````
```ad-error
There is a known issue coming with a future update to Obsidian (Front Matter upgrade to Properties).  

If you still wish to use Banners then you should save the following code into a new *.css file per: [How To - Install Custom CSS](https://obsidianttrpgtutorials.com/Obsidian+TTRPG+Tutorials/Tutorials/Basic+Tool+Usage/How+To+-+Install+Custom+CSS)  

This will fix it so that the Banner does not cover the new Properties UI. [See Video](https://youtu.be/2nFBteVz8Lg) for details on this issue.

~~~css
.obsidian-banner { position: unset; }
.obsidian-banner-wrapper { margin-top: 0; }
~~~
```

> [!error] 
> There is a known issue coming with a future update to Obsidian (Front Matter upgrade to Properties).  
> 
> If you still wish to use Banners then you should save the following code into a new *.css file per: [How To - Install Custom CSS](https://obsidianttrpgtutorials.com/Obsidian+TTRPG+Tutorials/Tutorials/Basic+Tool+Usage/How+To+-+Install+Custom+CSS)
> 
> This will fix it so that the Banner does not cover the new Properties UI. [See Video](https://youtu.be/2nFBteVz8Lg) for details on this issue.
>
>```css
>.obsidian-banner { position: unset; }
>.obsidian-banner-wrapper { margin-top: 0; }
>```
````

## Expected:

![](https://github.com/KosmosisDire/obsidian-webpage-export/assets/47476146/ae665ce1-7e7c-44f1-8a16-1fdc69ed7e27)

## What I got:

![](https://github.com/KosmosisDire/obsidian-webpage-export/assets/47476146/5954735e-3349-4391-953f-726380033863)

I can temporary fix it for myself by adding this css code directlty into html style block:
```css
.callout-icon.admonition-title-icon {
    color: aqua;
    width: 12px;
}
```

And I get not perfect but acceptable result:

![](https://github.com/KosmosisDire/obsidian-webpage-export/assets/47476146/bec84c92-93e8-43b8-b2a2-ba4415f26384)

P. S. I really enjoy your plugin, 'cause I always needed .html export. But I couldn't even imagine that good 👍

**Comment by KosmosisDire** [comment]
@Chupakabra0  I think the expected soluton to this would be to update to the new callout syntax. The Admonition plugin has an automated process for this in the settings. You could also try toggling the 'Use CSS snippets for custom callouts' setting. But I don't plan to add support for the old codeblock admonition style since it is only still supported for legacy reasons.

**Comment by Chupakabra0** [comment]
@KosmosisDire it's a pity. I really hate new callouts syntax. Old one is more readable especially when you want to put code in it.

Anyway, thank you for quick response and good luck.

**Comment by KosmosisDire** [comment]
I am sure you could get it to work because it is just a matter of getting whatever styles are used for it into the export. If you can find it and add it to a snippet or something it will work. But I am not sure how those codeblocks are styled.

**Comment by Chupakabra0** [comment]
After more accurate research i find out that callout svg .svg-icon has height and width attributes set to 24:

![center](https://github.com/KosmosisDire/obsidian-webpage-export/assets/47476146/ce5cbeb6-07d7-44db-bc0d-201efcc1a351)

But admonition svg .svg-inline--fa doesn't have no width, no height:

![](https://github.com/KosmosisDire/obsidian-webpage-export/assets/47476146/00e31ea4-4a31-4f74-9de9-4274feb4907d)

After adding width and height to it, it become perfectly works:

![](https://github.com/KosmosisDire/obsidian-webpage-export/assets/47476146/94e82e5f-b5c2-4088-8264-1e67da9be308)

Alas, I cannot add any html-attributes via my .css snippet. So my temporary hardcoded width/height to svg's parent div is still the easiest solution.

**Comment by KosmosisDire** [comment]
Huh weird that it had no width or height. If it's just that I would be able to fix it without having to add specific support for it. I'll test it and see what I can do

**Comment by vgyenge6** [comment]
Hi,

According to @Chupakabra0's comment https://github.com/KosmosisDire/obsidian-webpage-export/issues/284#issuecomment-1965497987 I also noticed many problems with this plugin after upgrading (unintentionally) to obsidian 1.5.8 and I also had to revert to the previous version 1.5.3. 
> Obsidian downgraded to 1.5.3 after 1.5.8 "had broken everything"
I'm using now Obsidian 1.5.3 and admonitions icons work well for me as I reported https://github.com/KosmosisDire/obsidian-webpage-export/issues/284#issuecomment-1956327194

**Comment by KosmosisDire** [comment]
> Hi,
> 
> According to @Chupakabra0's comment [#284 (comment)](https://github.com/KosmosisDire/obsidian-webpage-export/issues/284#issuecomment-1965497987) I also noticed many problems with this plugin after upgrading (unintentionally) to obsidian 1.5.8 and I also had to revert to the previous version 1.5.3.
> 
> > Obsidian downgraded to 1.5.3 after 1.5.8 "had broken everything"
> > I'm using now Obsidian 1.5.3 and admonitions icons work well for me as I reported [#284 (comment)](https://github.com/KosmosisDire/obsidian-webpage-export/issues/284#issuecomment-1956327194)

Yes #373

**Comment by vgyenge6** [comment]
thanks, I looked after #373. and found you released a newer 1.8.1b Pre-release where "Fix issue where windows would throw errors about path not found due to using forward slashes." also had resolved. I will try this release.
See: where I reported this issue https://github.com/KosmosisDire/obsidian-webpage-export/issues/366#issuecomment-1974791176
> There is also a little problem with the "/" in the path name even if I select the folder by the Browse button. e.g. D:/VALI/MDTest, on Windows I'm getting a Windows error message there is no such folder! Yes because the path separator on Windows is backslash. However, there is no problem with the saving of files to the local file system and the browser handles the path as File:///D:/VALI/MDTest/file.html
